src/coreclr/nativeaot/BuildIntegration/Microsoft.NETCore.Native.targets

```diff
@@ -126,26 +126,41 @@ The .NET Foundation licenses this file to you under the MIT license.
 
   <!-- The properties below need to be defined only after we've found the correct runtime package reference -->
   <Target Name="SetupProperties" DependsOnTargets="$(IlcSetupPropertiesDependsOn)" BeforeTargets="Publish">
+    <ItemGroup>
+      <_NETCoreAppFrameworkReference Include="@(ResolvedFrameworkReference)" Condition="'%(ResolvedFrameworkReference.RuntimePackName)' == 'Microsoft.NETCore.App.Runtime.NativeAOT.$(RuntimeIdentifier)'" />
+    </ItemGroup>
+
     <PropertyGroup>
       <!-- Define paths used in build targets to point to the runtime-specific ILCompiler implementation -->
       <IlcToolsPath Condition="'$(IlcToolsPath)' == ''">$(IlcHostPackagePath)\tools\</IlcToolsPath>
       <IlcSdkPath Condition="'$(IlcSdkPath)' == ''">$(RuntimePackagePath)\sdk\</IlcSdkPath>
-      <IlcFrameworkPath Condition="'$(IlcFrameworkPath)' == ''">$(RuntimePackagePath)\framework\</IlcFrameworkPath>
-      <IlcFrameworkNativePath Condition="'$(IlcFrameworkNativePath)' == ''">$(RuntimePackagePath)\framework\</IlcFrameworkNativePath>
+      <IlcFrameworkPath Condition="'$(IlcFrameworkPath)' == '' and '$(PublishAotUsingRuntimePack)' != 'true'">$(RuntimePackagePath)\framework\</IlcFrameworkPath>
+      <_NETCoreAppRuntimePackPath>%(_NETCoreAppFrameworkReference.RuntimePackPath)/runtimes/$(RuntimeIdentifier)/</_NETCoreAppRuntimePackPath>
+      <IlcFrameworkNativePath Condition="'$(IlcFrameworkNativePath)' == '' and '$(PublishAotUsingRuntimePack)' == 'true'">$(_NETCoreAppRuntimePackPath)\native\</IlcFrameworkNativePath>
+      <IlcFrameworkNativePath Condition="'$(IlcFrameworkNativePath)' == '' and '$(PublishAotUsingRuntimePack)' != 'true'">$(RuntimePackagePath)\framework\</IlcFrameworkNativePath>
+      <IlcFrameworkSdkPath Condition="'$(IlcFrameworkSdkPath)' == '' and '$(PublishAotUsingRuntimePack)' == 'true'">$(IlcFrameworkNativePath)</IlcFrameworkSdkPath>
+      <IlcFrameworkSdkPath Condition="'$(IlcFrameworkSdkPath)' == '' and '$(PublishAotUsingRuntimePack)' != 'true'">$(IlcSdkPath)</IlcFrameworkSdkPath>
       <IlcMibcPath Condition="'$(IlcMibcPath)' == ''">$(RuntimePackagePath)\mibc\</IlcMibcPath>
     </PropertyGroup>
 
-    <ItemGroup>
-      <PrivateSdkAssemblies Include="$(IlcSdkPath)*.dll" />
+    <ItemGroup Condition="'$(PublishAotUsingRuntimePack)' == 'true'">
+      <PrivateSdkAssemblies Include="$(IlcFrameworkSdkPath)*.dll"/>
+      <FrameworkAssemblies Include="@(RuntimePackAsset)" Condition="'%(Extension)' == '.dll'" />
+      <DefaultFrameworkAssemblies Include="@(FrameworkAssemblies)" />
```

Comment:
Why aren't `PrivateSdkAssemblies` included in `DefaultFrameworkAssemblies` in this case?

Reply:
It's not exactly intuitive but `RuntimePackAsset` already contain the files that end up in `PrivateSdkAssemblies`, so `FrameworkAssemblies` already contains the union of the two sets.

Reply:
Also, `FrameworkAssemblies` itself is not used beyond this point, so the additional assemblies included in it have no effect.

Reply:
> Also, `FrameworkAssemblies` itself is not used beyond this point, so the additional assemblies included in it have no effect.

Sorry, I got confused with this.
Aren't `FrameworkAssemblies` used through `DefaultFrameworkAssemblies` to populate `IlcReference`: https://github.com/dotnet/runtime/blob/ea94440d4224807d2a4afa19749ad40f2fb30c6e/src/coreclr/nativeaot/BuildIntegration/Microsoft.NETCore.Native.targets#L155

Reply:
`DefaultFrameworkAssemblies` is used beyond the point and it's expected to contain the union of `lib/net8.0/*.dll` and `native/*.dll` from the runtime pack.

In the case of `PublishAotUsingRuntimePack=false` these come from two different source directories and get merged into one item group (DefaultFrameworkAssemblies).

In the case of `PublishAotUsingRuntimePack=true` all the file in the DefaultFrameworkAssemblies item group come from the `RuntimePackAsset` asset group which already contain both the `lib/net8.0/*.dll` files and the `native/*.dll` files.

I was just trying to clarify that `FrameworkAssemblies` in the `PublishAotUsingRuntimePack=true` case will also include `PrivateSdkAssemblies` (native/*.dll), while in the `PublishAotUsingRuntimePack=false` case it won't include those files. However, `FrameworkAssemblies` itself is not used beyond this point, so the distinction doesn't matter.

Reply:
I can just rewrite it as to make it clearer:
```diff
-      <FrameworkAssemblies Include="@(RuntimePackAsset)" Condition="'%(Extension)' == '.dll'" />
-      <DefaultFrameworkAssemblies Include="@(FrameworkAssemblies)" />
+      <DefaultFrameworkAssemblies Include="@(RuntimePackAsset)" Condition="'%(Extension)' == '.dll'" />
```

Reply:
Thank you very much for the clarification! 

Regarding your last comment:
> I can just rewrite it as to make it clearer:

Would it make sense to keep having a single `ItemGroup` and just distinguish what gets included in `DefaultFrameworkAssemblies` based on  `PublishAotUsingRuntimePack` value?